### DISCUSSION TITLE
plot_2d_hist set log Z

### DISCUSSION THREAD

**Discussion by alozar-si** [cyrraz/plothist#119] [Q&A] [answered]
Hello, does plothist support setting log scale on Z axis for 2D hist?

**Comment by 0ctagon** [comment, marked as answer]
Hello!

Thank you for your question.
It does support this feature through the `pcolormesh_kwargs` and its `norm` argument (see [doc here](https://matplotlib.org/3.1.1/api/_as_gen/matplotlib.pyplot.pcolormesh.html)). You'll need to specify the color scale using a `LogNorm` ([doc here](https://matplotlib.org/stable/api/_as_gen/matplotlib.colors.LogNorm.html)).

If we take the `2d_hist_simple.py` example from the documentation, this is how it is implemented:

```
from plothist import get_dummy_data

df = get_dummy_data()

from plothist import make_2d_hist, plot_2d_hist
import matplotlib.pyplot as plt
from matplotlib.colors import LogNorm
import numpy as np

name_x = "variable_0"
name_y = "variable_1"

h = make_2d_hist([df[name_x], df[name_y]])

norm = LogNorm(vmin=1, vmax=10000) # Here we define the log scale. We could also use np.max(h.values()) to get the maximal value of h as the upper bin of the scale.

fig, ax, ax_colorbar = plot_2d_hist(
    h, colorbar_kwargs={"label": "Entries"}, pcolormesh_kwargs={"norm": norm} # Just add the norm argument to the pcolormesh_kwargs
)

ax.set_xlabel(name_x)
ax.set_ylabel(name_y)

ax.set_xlim(-10.5, 10)
ax.set_ylim(-9.3, 10)

fig.savefig("2d_hist_simple.pdf", bbox_inches="tight")
```


![Screenshot from 2024-02-21 19-00-38](https://github.com/cyrraz/plothist/assets/43090327/8a59b065-279e-45b6-b5a0-5f1e4dc33055)
